### DISCUSSION TITLE
Anyone doing Advent of Code (2024) in Koka?

### DISCUSSION THREAD

**Discussion by TheLostLambda** [koka-lang/koka#612] [General]
It could be enlightening (for me, an absolute Koka beginner) to see how others have solved things!

I'm not quite caught up, but as I go along, I'll be pushing things here: https://github.com/TheLostLambda/advent-of-code-2025

**Comment by TimWhiting** [comment]
Thanks for reminding me about this. I've wanted to do it in the past, and I think Koka is ready for it now (to some degree at least). 

https://github.com/TimWhiting/advent-2024

Feel free to join the unofficial community discord: https://discord.gg/fSsAcZtq

**Reply by TheLostLambda** [reply]
Nice! I'll join the Discord and follow along!

Also thanks for reminding me that it's 2024 and not 2025 currently... No idea how I got that turned around :sweat_smile:

**Comment by TheLostLambda** [comment]
I'll leave this open until Advent of Code is over, just so that more people stumble across it if they're interested :)

**Comment by HeikoRibberink** [comment]
I am currently doing [Advent of Code in Koka](https://github.com/HeikoRibberink/aoc20240)
It is quite a struggle, due to the poor error messages and documentation, the confusing type inference and the lack of proper support for array programming. Notably, I have no clue how to update a vector field of a struct, which I need for a proper matrix implementation.
